### ISSUE TITLE
can't see bounding boxes

### ISSUE THREAD

**Issue by Habibjackson** [theAIGuysCode/yolov4-deepsort#137]
I couldn't see any bounding box can anyone please suggest me a solution and also the info flag is not working.

**Comment by jamshaidsohail5** [comment]
Hi. I am facing the same issue on my local PC with GeForce RTX 3060 (6GB) but when I run the code on Goolge Colab, everything seems to work very much fine. If you are able to dig into the issue and find a solution please do share.

**Comment by jamshaidsohail5** [comment]
Hey I resolved the issue by upgrading the tensorflow version to 2.5.

pip install tensorflow-gpu==2.5

The above command did the job for me. Good luck.

**Comment by Habibjackson** [comment]
Thanks for your suggestion brother but i am already having tensorflow-gpu 2.8.0 still no bounding boxes.

**Comment by alantseone** [comment]
I also couldn't see any bounding box and I print the pred_bbox, only the first frame have output pred_bbox.
I had this problem before starting. 
QObject::moveToThread: Current thread (0x563bcd421950) is not the object's thread (0x563b8b6d2fb0).
Cannot move to target thread (0x563bcd421950)

I don't know it is affects the result?

**Comment by abdelmalek0** [comment]
tensorflow-gpu 2.8.0
yolov4-tiny
I couldn't see any bounding boxes

**Comment by abdelmalek0** [comment]
> tensorflow-gpu 2.8.0 yolov4-tiny I couldn't see any bounding boxes

I used tensorflow-gpu 2.3.0 and It worked finally

**Comment by Wetu-Vexo** [comment]
Hello guys, I also face the same issue with the gpu, but the code work fine on the cpu. So i try to convert the model with tensorflow cpu and run the object_tracking with tensorflow gpu.

**Comment by deshwalmahesh** [comment]
Yes, it is working fine with CPU. I there is an issue with the working of `GPU tensorflow` due to dependency

**Comment by kevin5k** [comment]
> Hello guys, I also face the same issue with the gpu, but the code work fine on the cpu. So i try to convert the model with tensorflow cpu and run the object_tracking with tensorflow gpu.

I confirm this method works and I managed to get the bounding box now!!! Specifically, I performed the _**save_model.py**_ command using _tf-cpu_ (i.e., tensorflow==2.3.0) environment. For real-time inference, I use the _tf-gpu_ (i.e., tensorflow==2.4.1) environment. Note that the libraries are created via anaconda environment only (i.e., no pip required).

Take the demo.avi as an example, the mean real-time inference performance I observed are 2fps (_tf-cpu_) as opposed to 22fps (_tf-gpu_) on a 2080Ti.

Hope this information helps everyone else.

**Comment by kevin5k** [comment]
> > tensorflow-gpu 2.8.0 yolov4-tiny I couldn't see any bounding boxes
> 
> I used tensorflow-gpu 2.3.0 and It worked finally

For the benefit of people like me, note the OS dependency differences for tensorflow-gpu package, I observed the following based on _conda-forge_ channel:
- (_Windows OS_) tensorflow-gpu -> {**_2.3.0_**, 2.5.0, 2.6.0} are available
- (_Linux OS - Ubuntu_) tensorflow-gpu -> {2.2.0, 2.4.1, 2.6, 2.7, 2.8} are available

**Comment by JDmoric** [comment]
> > Hello guys, I also face the same issue with the gpu, but the code work fine on the cpu. So i try to convert the model with tensorflow cpu and run the object_tracking with tensorflow gpu.
> 
> I confirm this method works and I managed to get the bounding box now!!! Specifically, I performed the _**save_model.py**_ command using _tf-cpu_ (i.e., tensorflow==2.3.0) environment. For real-time inference, I use the _tf-gpu_ (i.e., tensorflow==2.4.1) environment. Note that the libraries are created via anaconda environment only (i.e., no pip required).
> 
> Take the demo.avi as an example, the mean real-time inference performance I observed are 2fps (_tf-cpu_) as opposed to 22fps (_tf-gpu_) on a 2080Ti.
> 
> Hope this information helps everyone else.

Hello, I did as your tip that performed the save_model.py command using tf-cpu ( tensorflow==2.3.0) environment. For real-time inference, I use the tf-gpu (tensorflow-gpu==2.3.0) environment. But there is no detection.

How to deal with that.

Best wishes

**Comment by adityachintala** [comment]
> Hi. I am facing the same issue on my local PC with GeForce RTX 3060 (6GB) but when I run the code on Goolge Colab, everything seems to work very much fine. If you are able to dig into the issue and find a solution please do share.

Could you once state the tensorflow-gpu and opencv-python versions you were using back then to run the deepSORT model?

**Comment by adityachintala** [comment]
> > > Hello guys, I also face the same issue with the gpu, but the code work fine on the cpu. So i try to convert the model with tensorflow cpu and run the object_tracking with tensorflow gpu.
> > 
> > 
> > I confirm this method works and I managed to get the bounding box now!!! Specifically, I performed the _**save_model.py**_ command using _tf-cpu_ (i.e., tensorflow==2.3.0) environment. For real-time inference, I use the _tf-gpu_ (i.e., tensorflow==2.4.1) environment. Note that the libraries are created via anaconda environment only (i.e., no pip required).
> > Take the demo.avi as an example, the mean real-time inference performance I observed are 2fps (_tf-cpu_) as opposed to 22fps (_tf-gpu_) on a 2080Ti.
> > Hope this information helps everyone else.
> 
> Hello, I did as your tip that performed the save_model.py command using tf-cpu ( tensorflow==2.3.0) environment. For real-time inference, I use the tf-gpu (tensorflow-gpu==2.3.0) environment. But there is no detection.
> 
> How to deal with that.
> 
> Best wishes

Did you find the solution to this?
Even I'm facing the same issues.

**Comment by adityachintala** [comment]
> > tensorflow-gpu 2.8.0 yolov4-tiny I couldn't see any bounding boxes
> 
> I used tensorflow-gpu 2.3.0 and It worked finally

Opencv_python version?
